hazelcast/src/main/java/com/hazelcast/scheduledexecutor/impl/DistributedScheduledExecutorService.java

```diff
@@ -332,116 +310,54 @@ public String getQuorumName(final String name) {
-                for (Map.Entry<Integer, Map<String, Collection<ScheduledTaskDescriptor>>> partition : partitionsSnapshot
-                        .entrySet()) {
-
-                    int partitionId = partition.getKey();
-                    Map<String, Collection<ScheduledTaskDescriptor>> containers = partition.getValue();
-
-                    for (Map.Entry<String, Collection<ScheduledTaskDescriptor>> container : containers.entrySet()) {
-                        String containerName = container.getKey();
-                        Collection<ScheduledTaskDescriptor> tasks = container.getValue();
-
-                        int batchSize = getMergePolicyConfig(containerName).getBatchSize();
-                        SplitBrainMergePolicy mergePolicy = getMergePolicy(containerName);
-
-                        mergingEntries = new ArrayList<MergingEntryHolder<String, ScheduledTaskDescriptor>>();
-                        for (ScheduledTaskDescriptor descriptor : tasks) {
-                            MergingEntryHolder<String, ScheduledTaskDescriptor> mergingEntry
-                                    = createMergeHolder(nodeEngine.getSerializationService(), descriptor);
-                            mergingEntries.add(mergingEntry);
-                            size++;
-
-                            if (mergingEntries.size() == batchSize) {
-                                sendBatch(partitionId, containerName, mergePolicy, mergingEntries, mergeCallback);
-                                mergingEntries = new ArrayList<MergingEntryHolder<String, ScheduledTaskDescriptor>>(batchSize);
-                                operationCount++;
-                            }
-                        }
-
-                        tasks.clear();
-                        if (!mergingEntries.isEmpty()) {
-                            sendBatch(partitionId, containerName, mergePolicy, mergingEntries, mergeCallback);
-                            operationCount++;
-                        }
+            Map<Integer, Collection<ScheduledExecutorContainer>> containerMap = collector.getCollectedContainers();
+            for (Map.Entry<Integer, Collection<ScheduledExecutorContainer>> entry : containerMap.entrySet()) {
+                int partitionId = entry.getKey();
+                Collection<ScheduledExecutorContainer> containers = entry.getValue();
+
+                for (ScheduledExecutorContainer container : containers) {
+                    String name = container.getName();
+                    MergePolicyConfig mergePolicyConfig = collector.getMergePolicyConfig(container);
+                    SplitBrainMergePolicy mergePolicy = getMergePolicy(mergePolicyConfig);
+                    int batchSize = mergePolicyConfig.getBatchSize();
+
+                    mergingEntries = new ArrayList<MergingEntryHolder<String, ScheduledTaskDescriptor>>(batchSize);
+                    for (ScheduledTaskDescriptor descriptor : container.prepareForReplication(true).values()) {
```

Comment:
Now this is called after the call to `ScheduledExecutorService.reset()` but I think it works ok.